keps/sig-api-machinery/4977-dedicated-priority-level-for-event-requests/README.md

```diff
@@ -0,0 +1,982 @@
+This section is for explicitly listing the motivation, goals, and non-goals of
+this KEP.  Describe why the change is important and the benefits to users. The
+motivation section can optionally provide links to [experience reports] to
+demonstrate the interest in a KEP within the wider Kubernetes community.
+
+[experience reports]: https://github.com/golang/go/wiki/ExperienceReports
+-->
+
+Many features have been developed and use cases surfaced since the introduction
+of [KEP-1040]. These prompt us to revise APF default configuration to better
+make use of APF for better performance and reliability of API Server. One thing
+we have learned is that Event requests are plentiful in upscaling but they are
+resilient to unavailability of API Server. So we can deprioritize them when API
+Server is under stress.
+
+[KEP-1040]: https://github.com/kubernetes/enhancements/blob/97713189b3107b41c4c19505d04aa7ef22df063b/keps/sig-api-machinery/1040-priority-and-fairness/README.md
+
+### Goals
+
+<!--
+List the specific goals of the KEP. What is it trying to achieve? How will we
+know that this has succeeded?
+-->
+
+-   Better overload protection in perspective of whole Kubernetes control plane
+    instead of just API Server alone.
+-   More performant in cluster upscaling (e.g. increasing node count, deploying
+    pods)
+
+### Non-Goals
+
+<!--
+What is out of scope for this KEP? Listing non-goals helps to focus discussion
+and make progress.
+-->
+
+-   There will be no fundamental changes in APF's own implementation
+
+## Proposal
+
+<!--
+This is where we get down to the specifics of what the proposal actually is.
+This should have enough detail that reviewers can understand exactly what
+you're proposing, but should not include things like API designs or
+implementation. What is the desired outcome and how do we measure success?.
+The "Design Details" section below is for the real
+nitty-gritty.
+-->
+
+The APF feature provides flow control to API Server by introducing a limit on
+how much a priority level can consume the concurrency. In addition to that,
+borrowing is also instroduced to better utilize spare concurrency. The limits on
+borrowing are two-sided: a given priority level has a limit on how much it may
+borrow and a limit on how much may be borrowed from it. Worth mentioning, APF
+controller periodically populates current concurrency shares for each priority
+level (i.e. concurrency share that can be borrowed or need to be borrowed) by
+taking past usage into consideration. So in combination of those two aspects, a
+priority level's Nomial Share is only an initial value and the actual share
+might increase or decrease according to actual usage with borrowing. However,
+the owner priority level should have full Nominal Share as its current
+concurrency share if in full utilization.
+
+This KEP proposes the following settings:
+
+| Name  | Proposed Nominal Shares  | Proposed Lendable | Proposed Borrowing Limit | Proposed Guaranteed Shares |
```

Comment:
There are some attributes omitted here. You might compare how I filled them in in https://github.com/kubernetes/kubernetes/pull/128974

Reply:
Do you mean the weight and queue?

I can add weight if you believe the weighted borrow will be landed before this KEP, which I am more than happy to see. As for queue, I am inclined to make it reject right away given that the volume of events may easily overwhelm any queue size we will specify here.